ui/src/app/job-list/job-list-resolver.service.ts

```diff
@@ -21,11 +22,19 @@ export class JobListResolver implements Resolve<JobStream> {
 
   resolve(route: ActivatedRouteSnapshot,
```

Comment:
Man - the resolver runs again when you reattach to a component? That's pretty surprising.. I couldn't find any documentation one way or the other, so we'll have to base this on your local experimentation. Should double-check this again if you wind up changing reuseRoute per my below comment.

Reply:
Yup I was also surprised to find this...pretty annoying. Unfortunately `shouldReuseRoute` and all the other callbacks on `RouteReuseStrategy` are called after the resolver is triggered so we have to check the cache here.